src/linear_algebra/finite_dimensional.lean

```diff
@@ -0,0 +1,152 @@
+/-
+Copyright (c) 2019 Chris Hughes. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Hughes
+
+Definition and basic properties of finite dimensional vector spaces.
+
+The class `finite_dimensional` is defined to be `is_noetherian`, for ease of transfer of proofs.
```

Comment:
It might be a bit awkward at first, but shouldn't we just use `is_noetherian`? (Just like we shouldn't have `vector_space` in the first place.
@digama0 What do you think? Should this be a reducible def? Or an abbreviation? Or should it not be there at all?

Reply:
This is slightly different than how it's done with vector spaces. It's a reducible der, so you shouldn't have to define an instance when there's already an `is_noetherian` instance floating around.

Reply:
If we do use `is_noetherian` there definitely should be clear documentation of this somewhere, so people know where to look for finite dimensional vector spaces